srcs/app/http/refactor/HttResponseFactory.cpp

```diff
@@ -0,0 +1,97 @@
+#include "HttpResponseFactory.hpp"
+
+HttpResponseFactory::HttpResponseFactory(
+		const std::string raw_request,
+		const ServerConfig &server_config
+	): server_config_(server_config) {
+	request_ = new HttpRequestDTO(raw_request);
+	request_method_ = request_->GetMethod();
+	concreteResponses_.insert(std::make_pair(
+		"GET",
+		&HttpResponseFactory::createHttpGetResponse_
+	));
+	concreteResponses_.insert(std::make_pair(
+		"POST",
+		&HttpResponseFactory::createHttpPostResponse_
+	));
+	concreteResponses_.insert(std::make_pair(
+		"DELETE",
+		&HttpResponseFactory::createHttpDeleteResponse_
+	));
+}
+
+HttpResponseFactory::~HttpResponseFactory() {
+}
```

Comment:
We allocate the http methods from the concreteResponses_ map on the heap and never delete them. We can either delete them here or use the stack if it is possible.

Reply:
As said, this will be deleted from `Connection` class